pkg/system/phase4_configuring.go

```diff
@@ -597,6 +604,64 @@ func (r *Reconciler) prepareAzureBackingStore() error {
 	return nil
 }
 
+func (r *Reconciler) prepareGCPBackingStore() error {
+	// after we have cloud credential request, wait for credentials secret
+	cloudCredsSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.GCPCloudCreds.Spec.SecretRef.Name,
+			Namespace: r.GCPCloudCreds.Spec.SecretRef.Namespace,
+		},
+	}
+
+	util.KubeCheck(cloudCredsSecret)
+	if cloudCredsSecret.UID == "" {
+		// TODO: we need to figure out why secret is not created, and react accordingly
+		// e.g. maybe we are running on AWS but our CredentialsRequest is for GCP
+		r.Logger.Infof("Secret %q was not created yet by cloud-credentials operator. retry on next reconcile..", r.GCPCloudCreds.Spec.SecretRef.Name)
+		return fmt.Errorf("cloud credentials secret %q is not ready yet", r.GCPCloudCreds.Spec.SecretRef.Name)
+	}
+	r.Logger.Infof("Secret %s was created successfully by cloud-credentials operator", r.GCPCloudCreds.Spec.SecretRef.Name)
+	// create bucket in GCP
+	projectID := cloudCredsSecret.StringData["project_id"]
+	ctx := context.Background()
+	gcpclient, err := storage.NewClient(ctx)
+	if err != nil {
+		r.Logger.Info(err)
+	}
+	var bucketName = strings.ToLower(randname.GenerateWithPrefix("noobaacontainer", 5))
+	if err := r.createGCPBucketForBackingStore(gcpclient, projectID, bucketName); err != nil {
+		r.Logger.Info(err)
+	}
+	// create backing store
+	r.DefaultBackingStore.Spec.Type = nbv1.StoreTypeGoogleCloudStorage
+	r.DefaultBackingStore.Spec.GoogleCloudStorage = &nbv1.GoogleCloudStorageSpec{
+		TargetBucket: bucketName,
+		Secret: corev1.SecretReference{
+			Name:      cloudCredsSecret.Name,
+			Namespace: cloudCredsSecret.Namespace,
+		},
+	}
+	return nil
+}
+
+func (r *Reconciler) createGCPBucketForBackingStore(client *storage.Client, projectID, bucketName string) error {
+	// [START create_bucket]
+	ctx := context.Background()
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
+	defer cancel()
+	if err := client.Bucket(bucketName).Create(ctx, projectID, &storage.BucketAttrs{
```

Comment:
I wouldn't send BucketAttrs, just send nil and use the defaults.
Not sure if the majority customers would want to deploy low-cost storage in Asia.